libs/server/Resp/RespCommand.cs

```diff
@@ -125,201 +127,430 @@ internal sealed unsafe partial class RespServerSession : ServerSessionBase
-                    return RespCommand.GET;
-
-                // SET key1 value1 => [*3\r\n$3\r\nSET\r\n$]3\r\nkey\r\n$5\r\nvalue\r\n
-                if (*(long*)ptr == 724291344956994346L && *(2 + (int*)ptr) == 223626579 && *(ushort*)(12 + ptr) == 9226)
-                    return RespCommand.SET;
-
-                // setrange key offset value => [*4\r\n$8\r\nSETRANGE\r\nkey\r\noffset\r\nvalue]
-                if (*(long*)ptr == 724296842515133482L && *(long*)(ptr + 8) == 4992044754424579411L && *(ushort*)(ptr + 16) == 2573)
-                    return RespCommand.SETRANGE;
-
-                // GETRANGE key start length [*4\r\n$8\r\n GETSLICE \r\n key start legth]
-                if (*(long*)ptr == 724296842515133482L && *(long*)(ptr + 8) == 4992044754424579399L && *(ushort*)(ptr + 16) == 2573)
-                    return RespCommand.GETRANGE;
-
-                // PUBLISH channel message => [*3\r\n$7\r\nPUBLISH\r\n$]7\r\nchannel\r\n$7\r\message\r\n
-                if (*(long*)ptr == 724295743003505450L && *(2 + (int*)ptr) == 1279415632 && *(ushort*)(16 + ptr) == 9226)
-                    return RespCommand.PUBLISH;
-
-                // Single Key DEL key1 [*2\r\n$3\r\nDEL\r\n$]3\r\nkey\r\n
-                if (*(long*)ptr == 724291344956994090L && *(2 + (int*)ptr) == 223102276 && *(ushort*)(12 + ptr) == 9226)
-                    return RespCommand.DEL;
-
-                // INCR key1 ==> [*2\r\n$4\r\nINCR\r\n$]3\r\nkey\r\n
-                if (*(long*)ptr == 724292444468621866L && *(12 + (byte*)ptr) == 0x0d
-                    && *(2 + (int*)ptr) == 0x52434e49 && *(ushort*)(13 + ptr) == 9226)
-                    return RespCommand.INCR;
-
-                // Fast path for transactions with few arguments (command array size < 10)
-                // RUNTXP ==> [*X\r\n$6\r\nRUNTXP\r\n]
-                if (*(long*)(ptr + 4) == 6074886758213695012 && *(int*)(ptr + 12) == 168644696)
+                switch ((*(int*)ptr >> 8) & 0xFF)
                 {
-                    if (*ptr == '*' && *(ushort*)(ptr + 2) == 2573 && *(ptr + 1) >= '0' && *(ptr + 1) <= '9')
-                    {
-                        return RespCommand.RUNTXP;
-                    }
+                    case '0':
+
+                        if ((*(1 + (int*)ptr) & Read<int>(smallBulkStringMask)) == Read<int>(smallBulkStringMask))
+                        {
+                            switch ((*(1 + (int*)ptr) >> 8) & 0xFF)
+                            {
+                                case '6':
+                                    if (bytesRead - readHead >= 16) // TODO: make sure this condition is appropriate per command
+                                    {
+                                        if (*(1 + (long*)ptr) == Read<long>("RUNTXP\r\n"u8))
+                                            return RespCommand.RUNTXP;
+                                    }
+
+                                    break;
```

Comment:
Some variation of following should be possible to reduce indentation. Roslyn should lower to same conditions
```suggestion
                                case '6' when bytesRead - readHead >= 16:
                                    if (*(1 + (long*)ptr) == Read<long>("RUNTXP\r\n"u8))
                                        return RespCommand.RUNTXP;
```

Reply:
Thanks! Realized I could take this to the extreme actually x)